wolfcrypt/src/ed25519.c

```diff
@@ -1069,18 +1097,36 @@ int wc_ed25519_import_private_key(const byte* priv, word32 privSz,
         return BAD_FUNC_ARG;
     }
 
+    XMEMCPY(key->k, priv, ED25519_KEY_SIZE);
+    key->privKeySet = 1;
+
     /* import public key */
-    ret = wc_ed25519_import_public(pub, pubSz, key);
+    ret = wc_ed25519_import_public_ex(pub, pubSz, key, trusted);
     if (ret != 0)
         return ret;
 
     /* make the private key (priv + pub) */
-    XMEMCPY(key->k, priv, ED25519_KEY_SIZE);
     XMEMCPY(key->k + ED25519_KEY_SIZE, key->p, ED25519_PUB_KEY_SIZE);
```

Comment:
This is already done in `wc_ed25519_import_public_ex`.

Reply:
@julek-wolfssl I do not see where `wc_ed25519_import_public_ex` is guaranteed to put the public key into `key->k + ED25519_KEY_SIZE`. I don't think we should remove this line.

Reply:
I was wrong. I didn't read it correctly.